src/main/kotlin/com/exactpro/th2/codec/oracle/logminer/LogMinerTransformer.kt

```diff
@@ -138,15 +136,430 @@ class LogMinerTransformer(private val config: LogMinerConfiguration) : IPipeline
         }
     }
 
+    // TODO: handle parsing exception
+    class InsertListener(private val builder: MapBuilder<String, Any?>, private val prefix: String) : PlSqlParserBaseListener() {
+        private val columNames: Queue<String> = LinkedList()
+        private var stage = Stage.BEGIN
+        private val expressionHolder: MutableList<Expression> = mutableListOf()
+
+        override fun visitErrorNode(node: ErrorNode) {
+            super.visitErrorNode(node)
+        }
+
+        override fun enterInsert_statement(ctx: PlSqlParser.Insert_statementContext) {
+            LOGGER.trace { "enterInsert_statement token index: ${ctx.start.tokenIndex}, text: ${ctx.text}" }
+
+            stage = Stage.BEGIN
+            columNames.clear()
+            expressionHolder.clear()
+        }
+
+        override fun enterInsert_into_clause(ctx: PlSqlParser.Insert_into_clauseContext) {
+            LOGGER.trace { "enterInsert_into_clause token index: ${ctx.start.tokenIndex}, text: ${ctx.text}" }
+            check(stage == Stage.BEGIN) {
+                "Incorrect stage for parsing insert into clause, expected: ${Stage.BEGIN}, actual: $stage, text: ${ctx.text}"
+            }
```

Comment:
I think this can be extracted into a method to avoid duplication in almost all methods

Reply:
Yes, I will working on it later